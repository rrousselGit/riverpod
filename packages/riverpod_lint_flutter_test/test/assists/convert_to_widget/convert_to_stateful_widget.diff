Message: `Convert to StatefulWidget`
Priority: 30
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:25`:
```
}

- class Hook extends HookWidget {
-   const Hook({super.key});
- 
-   @override
+ class Hook extends StatefulWidget {
+   const Hook({super.key});
+ 
+   @override
+   State<Hook> createState() => _HookState();
+ }
+ 
+ class _HookState extends State<Hook> {
@override
  Widget build(BuildContext context) {
```
---
Message: `Convert to StatefulWidget`
Priority: 30
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:34`:
```
}

- class HookConsumer extends HookConsumerWidget {
-   const HookConsumer({super.key});
- 
-   @override
-   Widget build(
-     BuildContext context,
-     WidgetRef ref,
-   ) {
+ class HookConsumer extends StatefulWidget {
+   const HookConsumer({super.key});
+ 
+   @override
+   State<HookConsumer> createState() => _HookConsumerState();
+ }
+ 
+ class _HookConsumerState extends State<HookConsumer> {
+ @override
+   Widget build(
+     BuildContext context) {
    return const Placeholder();
  }
```
---
Message: `Convert to StatefulWidget`
Priority: 30
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:78`:
```
}

- class HookStateful extends StatefulHookWidget {
+ class HookStateful extends StatefulWidget {
  const HookStateful({super.key});

```
---
Message: `Convert to StatefulWidget`
Priority: 30
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:92`:
```
}

- class ConsumerStateful extends ConsumerStatefulWidget {
-   const ConsumerStateful({super.key});
- 
-   @override
-   ConsumerState<ConsumerStateful> createState() => _ConsumerStatefulState();
- }
- 
- class _ConsumerStatefulState extends ConsumerState<ConsumerStateful> {
+ class ConsumerStateful extends StatefulWidget {
+   const ConsumerStateful({super.key});
+ 
+   @override
+   State<ConsumerStateful> createState() => _ConsumerStatefulState();
+ }
+ 
+ class _ConsumerStatefulState extends State<ConsumerStateful> {
  @override
  Widget build(BuildContext context) {
```
---
Message: `Convert to StatefulWidget`
Priority: 30
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:106`:
```
}

- class HookConsumerStateful extends StatefulHookConsumerWidget {
-   const HookConsumerStateful({super.key});
- 
-   @override
-   ConsumerState<HookConsumerStateful> createState() =>
-       _HookConsumerStatefulState();
- }
- 
- class _HookConsumerStatefulState extends ConsumerState<HookConsumerStateful> {
+ class HookConsumerStateful extends StatefulWidget {
+   const HookConsumerStateful({super.key});
+ 
+   @override
+   State<HookConsumerStateful> createState() =>
+       _HookConsumerStatefulState();
+ }
+ 
+ class _HookConsumerStatefulState extends State<HookConsumerStateful> {
  @override
  Widget build(BuildContext context) {
```
---
Message: `Convert to StatefulWidget`
Priority: 30
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:121`:
```
}

- class Consumer extends ConsumerWidget {
-   const Consumer({super.key});
- 
-   @override
-   Widget build(
-     BuildContext context,
-     WidgetRef ref,
-   ) {
+ class Consumer extends StatefulWidget {
+   const Consumer({super.key});
+ 
+   @override
+   State<Consumer> createState() => _ConsumerState();
+ }
+ 
+ class _ConsumerState extends State<Consumer> {
+ @override
+   Widget build(
+     BuildContext context) {
    return const Placeholder();
  }
```
---
Message: `Convert to StatefulWidget`
Priority: 30
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:153`:
```
}

- class HookConsumerWithField extends HookConsumerWidget {
-   const HookConsumerWithField({
-     super.key,
-     required this.field,
-   });
- 
-   final int field;
-   static final int staticField = 42;
- 
-   @override
-   Widget build(
-     BuildContext context,
-     WidgetRef ref,
-   ) {
-     return Column(
-       children: [
-         Text('$field'),
-         Text('$staticField'),
+ class HookConsumerWithField extends StatefulWidget {
+   const HookConsumerWithField({
+     super.key,
+     required this.field,
+   });
+ 
+   final int field;
+   static final int staticField = 42;
+ 
+   @override
+   State<HookConsumerWithField> createState() => _HookConsumerWithFieldState();
+ }
+ 
+ class _HookConsumerWithFieldState extends State<HookConsumerWithField> {
+ @override
+   Widget build(
+     BuildContext context) {
+     return Column(
+       children: [
+         Text('${widget.field}'),
+         Text('${HookConsumerWithField.staticField}'),
      ],
    );
```
---
Message: `Convert to StatefulWidget`
Priority: 30
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:180`:
```
}

- class ConsumerStatefulWithField extends ConsumerStatefulWidget {
-   const ConsumerStatefulWithField({
-     super.key,
-     required this.field,
-     required this.foo,
-   });
- 
-   final int field;
-   final FooClass foo;
-   static final int staticField = _constantNumber;
- 
-   @override
-   ConsumerState<ConsumerStatefulWithField> createState() =>
-       _ConsumerStatefulWithFieldState();
- }
- 
- const _constantNumber = 42;
- 
- class _ConsumerStatefulWithFieldState
-     extends ConsumerState<ConsumerStatefulWithField> {
+ class ConsumerStatefulWithField extends StatefulWidget {
+   const ConsumerStatefulWithField({
+     super.key,
+     required this.field,
+     required this.foo,
+   });
+ 
+   final int field;
+   final FooClass foo;
+   static final int staticField = _constantNumber;
+ 
+   @override
+   State<ConsumerStatefulWithField> createState() =>
+       _ConsumerStatefulWithFieldState();
+ }
+ 
+ const _constantNumber = 42;
+ 
+ class _ConsumerStatefulWithFieldState
+     extends State<ConsumerStatefulWithField> {
  void printFoo() {
    print(widget.foo);
```
---
