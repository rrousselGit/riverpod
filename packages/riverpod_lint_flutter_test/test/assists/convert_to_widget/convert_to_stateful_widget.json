[{"priority":30,"change":{"message":"Convert to StatefulWidget","edits":[{"fileStamp":0,"edits":[{"offset":563,"length":0,"replacement":"@override\n  State<Hook> createState() => _HookState();\n}\n\nclass _HookState extends State<Hook> {\n"},{"offset":520,"length":10,"replacement":"StatefulWidget"}]}],"linkedEditGroups":[]}},{"priority":30,"change":{"message":"Convert to StatefulWidget","edits":[{"fileStamp":0,"edits":[{"offset":787,"length":23,"replacement":""},{"offset":737,"length":0,"replacement":"@override\n  State<HookConsumer> createState() => _HookConsumerState();\n}\n\nclass _HookConsumerState extends State<HookConsumer> {\n"},{"offset":678,"length":18,"replacement":"StatefulWidget"}]}],"linkedEditGroups":[]}},{"priority":30,"change":{"message":"Convert to StatefulWidget","edits":[{"fileStamp":0,"edits":[{"offset":1708,"length":19,"replacement":"State<HookStateful>"},{"offset":1615,"length":19,"replacement":"State<HookStateful>"},{"offset":1544,"length":18,"replacement":"StatefulWidget"}]}],"linkedEditGroups":[]}},{"priority":30,"change":{"message":"Convert to StatefulWidget","edits":[{"fileStamp":0,"edits":[{"offset":2045,"length":31,"replacement":"State<ConsumerStateful>"},{"offset":1930,"length":31,"replacement":"State<ConsumerStateful>"},{"offset":1851,"length":22,"replacement":"StatefulWidget"}]}],"linkedEditGroups":[]}},{"priority":30,"change":{"message":"Convert to StatefulWidget","edits":[{"fileStamp":0,"edits":[{"offset":2424,"length":35,"replacement":"State<HookConsumerStateful>"},{"offset":2291,"length":35,"replacement":"State<HookConsumerStateful>"},{"offset":2204,"length":26,"replacement":"StatefulWidget"}]}],"linkedEditGroups":[]}},{"priority":30,"change":{"message":"Convert to StatefulWidget","edits":[{"fileStamp":0,"edits":[{"offset":2676,"length":23,"replacement":""},{"offset":2626,"length":0,"replacement":"@override\n  State<Consumer> createState() => _ConsumerState();\n}\n\nclass _ConsumerState extends State<Consumer> {\n"},{"offset":2575,"length":14,"replacement":"StatefulWidget"}]}],"linkedEditGroups":[]}}]