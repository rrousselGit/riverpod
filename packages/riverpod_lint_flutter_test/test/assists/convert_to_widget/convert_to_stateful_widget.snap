Message: `Convert to StatefulWidget`
<<<< start: 520 -- end: 562
HookWidget {
  const Hook({super.key});

 
==== start: 520 -- end: 663
StatefulWidget {
  const Hook({super.key});

  @override
  State<Hook> createState() => _HookState();
}

class _HookState extends State<Hook> {
>>>>
Message: `Convert to StatefulWidget`
<<<< start: 678 -- end: 809
HookConsumerWidget {
  const HookConsumer({super.key});

  @override
  Widget build(
    BuildContext context,
    WidgetRef ref,
 
==== start: 678 -- end: 911
StatefulWidget {
  const HookConsumer({super.key});

  @override
  State<HookConsumer> createState() => _HookConsumerState();
}

class _HookConsumerState extends State<HookConsumer> {
@override
  Widget build(
    BuildContext contex
>>>>
Message: `Convert to StatefulWidget`
<<<< start: 1552 -- end: 1556
Hook
==== start: 1552 -- end: 1552
>>>>
Message: `Convert to StatefulWidget`
<<<< start: 1851 -- end: 2052
ConsumerStatefulWidget {
  const ConsumerStateful({super.key});

  @override
  ConsumerState<ConsumerStateful> createState() => _ConsumerStatefulState();
}

class _ConsumerStatefulState extends Consume
==== start: 1851 -- end: 2028
StatefulWidget {
  const ConsumerStateful({super.key});

  @override
  State<ConsumerStateful> createState() => _ConsumerStatefulState();
}

class _ConsumerStatefulState extends
>>>>
Message: `Convert to StatefulWidget`
<<<< start: 2212 -- end: 2431
HookConsumerWidget {
  const HookConsumerStateful({super.key});

  @override
  ConsumerState<HookConsumerStateful> createState() =>
      _HookConsumerStatefulState();
}

class _HookConsumerStatefulState extends Consume
==== start: 2212 -- end: 2403
Widget {
  const HookConsumerStateful({super.key});

  @override
  State<HookConsumerStateful> createState() =>
      _HookConsumerStatefulState();
}

class _HookConsumerStatefulState extends
>>>>
Message: `Convert to StatefulWidget`
<<<< start: 2575 -- end: 2698
ConsumerWidget {
  const Consumer({super.key});

  @override
  Widget build(
    BuildContext context,
    WidgetRef ref,
 
==== start: 2575 -- end: 2788
StatefulWidget {
  const Consumer({super.key});

  @override
  State<Consumer> createState() => _ConsumerState();
}

class _ConsumerState extends State<Consumer> {
@override
  Widget build(
    BuildContext contex
>>>>
